haystack/modeling/data_handler/data_silo.py

```diff
@@ -136,61 +113,22 @@ def _get_dataset(self, filename: Optional[Union[str, Path]], dicts: Optional[Lis
                         random.shuffle(dicts)
 
         num_dicts = len(dicts)
-        multiprocessing_chunk_size, num_cpus_used = calc_chunksize(
-            num_dicts=num_dicts, max_processes=self.max_processes, max_chunksize=self.max_multiprocessing_chunksize
-        )
-
-        with ExitStack() as stack:
-            if self.max_processes > 1:  # use multiprocessing only when max_processes > 1
-                if self.multiprocessing_strategy:
-                    if self.multiprocessing_strategy in mp.get_all_sharing_strategies():
-                        mp.set_sharing_strategy(self.multiprocessing_strategy)
-                    else:
-                        logger.warning(
-                            f"{self.multiprocessing_strategy} is unavailable, "
-                            f"falling back to default multiprocessing sharing strategy of your OS."
-                        )
-
-                p = stack.enter_context(mp.Pool(processes=num_cpus_used))
-
-                logger.info(
-                    f"Got ya {num_cpus_used} parallel workers to convert {num_dicts} dictionaries "
-                    f"to pytorch datasets (chunksize = {multiprocessing_chunk_size})..."
-                )
-                log_ascii_workers(num_cpus_used, logger)
-
-                results = p.imap(
-                    partial(self._dataset_from_chunk, processor=self.processor),
-                    grouper(dicts, multiprocessing_chunk_size),
-                    chunksize=1,
-                )
-            else:
-                logger.info(
-                    f"Multiprocessing disabled, using a single worker to convert {num_dicts}"
-                    f"dictionaries to pytorch datasets."
-                )
+        datasets = []
+        problematic_ids_all = set()
+        batch_size = self.max_multiprocessing_chunksize
+        for i in tqdm(range(0, num_dicts, batch_size), desc="Preprocessing dataset", unit=" Dicts"):
+            processing_batch = dicts[i : i + batch_size]
+            indices = [i for i in range(len(processing_batch))]  # indices is a remnant from multiprocessing era
```

Comment:
I think you can also use 
```python
indices = list(range(len(processing_batch)))
```
which is a bit faster than the list comprehension for this use case.